examples/AnimalAndFlower/model.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+import torch.nn.functional as F
+import torch
+from torch import nn
+from tqdm import tqdm
+import torch.optim as optim
+
+from regr.program import SolverPOIProgram, IMLProgram
+from regr.program.metric import MacroAverageTracker, PRF1Tracker, DatanodeCMMetric
+from regr.program.loss import BCEWithLogitsIMLoss
+from graph import graph
+from regr.sensor.pytorch.sensors import ReaderSensor
+from regr.sensor.pytorch.learners import ModuleLearner
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+output_size = 45
+
+
+class ImageNetwork(torch.nn.Module):
+    def __init__(self, n_outputs=2):
+        super(ImageNetwork, self).__init__()
+        self.num_outputs = n_outputs
+        self.conv1 = nn.Conv2d(3, 6, 5)
+        self.pool = nn.MaxPool2d(2, 2)
+        self.conv2 = nn.Conv2d(6, 16, 5)
+        self.conv3 = nn.Conv2d(16, 5, 5)
+        self.conv4 = nn.Conv2d(5, 5, 3)
+        self.bn = nn.BatchNorm2d(5)
+
```

Comment:
the exact dimensions should be clarified here, I think this is a disadvantage of the framework. Can we change this at all? to interpret the dimensions from the data graph?

Reply:
if you change the model, then you have the change the output_size too.